src/main/cpp/src/from_json_to_structs.cu

```diff
@@ -0,0 +1,1100 @@
+
+  auto output = cudf::make_fixed_width_column(
+    cudf::data_type{cudf::type_id::BOOL8}, string_count, cudf::mask_state::UNALLOCATED, stream, mr);
+  auto validity = rmm::device_uvector<bool>(string_count, stream);
+
+  auto const input_sv = cudf::strings_column_view{input};
+  auto const offsets_it =
+    cudf::detail::offsetalator_factory::make_input_iterator(input_sv.offsets());
+  auto const d_input_ptr = cudf::column_device_view::create(input, stream);
+  auto const is_valid_it = cudf::detail::make_validity_iterator<true>(*d_input_ptr);
+  auto const output_it   = thrust::make_zip_iterator(
+    thrust::make_tuple(output->mutable_view().begin<bool>(), validity.begin()));
+  thrust::tabulate(
+    rmm::exec_policy_nosync(stream),
+    output_it,
+    output_it + string_count,
+    [chars = input_sv.chars_begin(stream), offsets = offsets_it, is_valid = is_valid_it] __device__(
+      auto idx) -> thrust::tuple<bool, bool> {
+      if (is_valid[idx]) {
+        auto const start_offset = offsets[idx];
+        auto const end_offset   = offsets[idx + 1];
+        auto const size         = end_offset - start_offset;
+        auto const str          = chars + start_offset;
+
+        if (size == 4 && str[0] == 't' && str[1] == 'r' && str[2] == 'u' && str[3] == 'e') {
+          return {true, true};
+        }
+        if (size == 5 && str[0] == 'f' && str[1] == 'a' && str[2] == 'l' && str[3] == 's' &&
+            str[4] == 'e') {
+          return {false, true};
+        }
+      }
+
+      // Either null input, or the input string is neither `true` nor `false`.
+      return {false, false};
+    });
+
+  auto [null_mask, null_count] =
+    cudf::detail::valid_if(validity.begin(), validity.end(), thrust::identity{}, stream, mr);
+  output->set_null_mask(null_count > 0 ? std::move(null_mask) : rmm::device_buffer{0, stream, mr},
+                        null_count);
+
+  return output;
+}
+
+std::unique_ptr<cudf::column> cast_strings_to_integers(cudf::column_view const& input,
+                                                       cudf::data_type output_type,
+                                                       rmm::cuda_stream_view stream,
+                                                       rmm::device_async_resource_ref mr)
+{
+  CUDF_FUNC_RANGE();
+
+  auto const string_count = input.size();
+  if (string_count == 0) { return cudf::make_empty_column(output_type); }
+
+  auto const input_sv = cudf::strings_column_view{input};
+  auto const input_offsets_it =
+    cudf::detail::offsetalator_factory::make_input_iterator(input_sv.offsets());
+  auto const d_input_ptr = cudf::column_device_view::create(input, stream);
+  auto const is_valid_it = cudf::detail::make_validity_iterator<true>(*d_input_ptr);
+
+  // We need to nullify the invalid string rows.
+  // Technically, we should just mask out these rows as invalid and ignore them.
+  // However, `spark_rapids_jni::string_to_integer` cannot handle these non-empty null rows,
```

Comment:
Should we make an issue to fix that?

Reply:
Oh wait, maybe I was wrong. I'll check this again.

Reply:
Indeed I've seen a bug in `string_to_decimal_kernel` that cannot handle non-empty null rows, not `string_to_integer`. I'll check for `string_to_integer` more carefully tomorrow. The bug in `string_to_decimal_kernel` also needs to be fixed so we can leverage non-empty null handling in this PR.

Reply:
https://github.com/NVIDIA/spark-rapids-jni/pull/2607